src/Users/Admin/admAccountsPage.js

```diff
@@ -0,0 +1,199 @@
+  const [fileName, setFile] = useState('')
+  const [rolesList, setRolesList] = useState([])
+  const [errMessage, setErrMessage] = useState('')
+  const singleCreationUrl = process.env.REACT_APP_BACKEND_URL + '/adm/register'
+  const csvCreationUrl = process.env.REACT_APP_BACKEND_URL + '/adm/csvRegisterUser'
+
+  useEffect(() => {
+    const rolesUrl = process.env.REACT_APP_BACKEND_URL + '/adm/rolesList'
+
+    try {
+      fetch(rolesUrl, {
+        method: 'GET',
+        headers: {
+          'x-auth-token': sessionStorage.getItem('token'),
+          'Content-Type': 'application/json'
+        }
+      }).then(response => response.json())
+        .then(data => setRolesList(data.roles))
+        .catch(error => setErrMessage(error.message))
+    } catch (e) {
+      setErrMessage(e.message)
+    }
+  }, [])
+
+  const handleSingleAccount = () => {
+    setIsOpenSingle(!isOpenSingle)
+    setFirstName('')
+    setLastName('')
+    setEmail('')
+    setErrMessage('')
+    if (isOpenMany) {
+      setIsOpenMany(false)
+    }
+  }
+
+  const handleManyAccounts = () => {
+    setIsOpenMany(!isOpenMany)
+    setFile()
+    setErrMessage('')
+    if (isOpenSingle) {
+      setIsOpenSingle(false)
+    }
+  }
+
+  const handleFirstNameChange = (event) => {
+    setFirstName(event.target.value)
+  }
+
+  const handleLastNameChange = (event) => {
+    setLastName(event.target.value)
+  }
+
+  const handleEmailChange = (event) => {
+    setEmail(event.target.value)
+  }
+
+  const handleFileChange = (event) => {
+    setFile(event.target.files[0])
+  }
+
+  const singleAccountCreation = async (event) => {
+    event.preventDefault()
+
+    const filteredArray = rolesList.filter(item => item.levelOfAccess === 2)
+    const roleId = filteredArray.map(item => item._id)
+
+    const payload = {
+      firstname,
+      lastname,
+      email,
+      role: roleId[0],
+      classes: []
+    }
+
+    try {
+      await fetch(singleCreationUrl, {
+        method: 'POST',
+        headers: {
+          'x-auth-token': sessionStorage.getItem('token'),
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(payload)
+      }).then(response => {
+        if (response.ok) {
+          window.location.reload()
+        } else {
+          const data = response.json()
+
+          setErrMessage(data.message)
+        }
+      })
+    } catch (e) {
+      setErrMessage(e.message)
+    }
+  }
+
+  const csvAccountCreation = async (event) => {
+    event.preventDefault()
+    const formData = new FormData()
+
+    formData.append('csv', fileName)
+
+    try {
+      await fetch(csvCreationUrl, {
+        method: 'POST',
+        headers: {
+          'x-auth-token': sessionStorage.getItem('token')
+        },
+        body: formData
+      }).then(response => {
+        if (response.ok) {
+          setErrMessage('this is fine')
+        } else {
+          const data = response.json()
+
+          setErrMessage(data.message)
+        }
```

Comment:
?

Reply:
ErrMessage n'est pas que pour les messages d'erreur mais c'est vrai que je pourrai mettre autre chose que 'this is fine'